### DISCUSSION TITLE
Configs related to read path

### DISCUSSION THREAD

**Discussion by madhaviuddaraju** [cortexproject/cortex#5315] [General]
I would appreciate it if someone could provide me with some advice or best practices regarding the configurations that should be checked in the read path, including Store gateways, Ingester, Cache, Querier, and Query-frontend.
Based on my observation, I have noticed that long-duration range queries beyond 1 week or 1 month tend to have higher response times. Additionally, I have also observed that response times are high when dealing with large data result sets.

Can you please let us know the related configs to improve cluster performance ?

**Comment by yeya24** [comment]
> Based on my observation, I have noticed that long-duration range queries beyond 1 week or 1 month tend to have higher response times. Additionally, I have also observed that response times are high when dealing with large data result sets.

That's correct and we observed the same thing. Unfortunately there is not much config we can tune right now for better query performance but we do have plans to improve such situations.

What you can do currently is probably to make sure your components are set up correctly with enough resources, your blocks are compacted in time, etc.
